### PR TITLE
Automated people data purge

### PR DESCRIPTION
**Pull request by TheWalkingLeek**
Closes: #2106 

Also merge: https://github.com/hitobito/hitobito_youth/pull/36

### CODE REVIEW

File: db/migrate/20230926105011_add_minimized_at_to_people.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#  Copyright (c) 2023, Pfadibewegung Schweiz. This file is part of
+#  hitobito_pbs and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_pbs.
```
Comment (mtnstar) [inline db/migrate/20230926105011_add_minimized_at_to_people.rb:4]:
```suggestion
#  https://github.com/hitobito/hitobito.
```

File: app/domain/people/minimizer.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2023, Pfadibewegung Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
```
Comment (mtnstar) [inline app/domain/people/minimizer.rb:7]: maybe leave a comment what this class is meant for?

File: app/domain/people/minimizer.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+class People::Minimizer
+
+  def initialize(person)
+    @person = person
+  end
+
+  def run
+    # noop - override in wagon
```
Comment (mtnstar) [inline app/domain/people/minimizer.rb:15]: update minimized_at in here instead of further up?

File: app/models/role.rb [hunk 4]
```diff
@@ -248,4 +249,8 @@ def prevent_changes
 
     raise ActiveRecord::ReadOnlyRecord unless new_record? || only_archival
   end
+
+  def reset_person_minimized_at
+    person.minimized_at = nil
```
Comment (mtnstar) [inline app/models/role.rb:254]: doesn't it require a person.update! ?
there's no spec for this case, is there?

File: config/settings.yml [hunk 5]
```diff
@@ -144,6 +144,10 @@ people:
   default_sort: name
   abos: true
   totp_drift: 15
+  cleanup_cutoff_duration:
+    regarding_current_sign_in_at: 1000 # months
+    regarding_roles: 1000 # months
+    regarding_event_participations: 100 # years
```
Comment (mtnstar) [inline config/settings.yml:150]:
```suggestion
    regarding_event_participation_years: 100
```

would it make sense to mention the unit (months, years) in the key?

**Review by mtnstar — APPROVED** [review]
functionality seams to be pretty good :partying_face: 
some comments to check

File: app/domain/people/minimizer.rb (hunk 3 above)
Comment (TheWalkingLeek) [reply app/domain/people/minimizer.rb:15]: Yes! I realized this was necessary in the follow up ticket too so I'll change it here :)

**Pull request merged**